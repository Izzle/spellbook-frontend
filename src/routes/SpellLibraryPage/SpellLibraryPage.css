.SpellLibraryPage__container {
    width: 600px;
    margin: 0 auto;
}

.SpellLibraryPage__Spells__container {
    display: flex;
}