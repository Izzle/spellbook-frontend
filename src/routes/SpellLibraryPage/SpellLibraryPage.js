import React, { Component } from 'react';
import SpellContext from '../../contexts/SpellContext';
import SpellLibraryOptions from '../../components/SpellLibraryOptions/SpellLibraryOptions';
import Spell from '../../components/Spell/Spell';
import STORE from '../../store';
import './SpellLibraryPage.css';

export default class SpellLibraryPage extends Component {
  static contextType = SpellContext;
  
  componentDidMount() {
    // Use something like this when the API is working
    //   SpellApiService.getSpells()
    //    .then(this.context.setSpellBookList)
    //    .catch(this.context.setError);
    this.context.setSpells(STORE.spells);
  }

  render() {
    const { spells = [] } = this.context;

    return (
      <section className='SpellLibraryPage'>
        <div className='SpellLibraryPage__container'>
          <SpellLibraryOptions />
          <div className='SpellLibraryPage__Spells__container'>
            {spells.map((spell, idx) => {
                return <Spell
                  key={idx}
                  id={spell.id}
                  name={spell.spell_name}
                  classes={spell.spell_classes}
                  level={spell.spell_level}
                  school={spell.spell_school}
                  range={spell.spell_range}
                  castingTime={spell.cast_time}
                  spellComponents={spell.spell_components}
                  duration={spell.spell_duration}
                  description={spell.spell_description}
                  higherLevels={spell.higherLevels}
                />
            })}
          </div>
        </div>
      </section>
    )
  }
}