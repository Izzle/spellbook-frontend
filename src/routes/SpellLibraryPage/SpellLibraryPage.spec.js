import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellLibraryPage from './SpellLibraryPage';
import { BrowserRouter } from 'react-router-dom';


describe('<SpellLibraryPage />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><SpellLibraryPage /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><SpellLibraryPage /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});