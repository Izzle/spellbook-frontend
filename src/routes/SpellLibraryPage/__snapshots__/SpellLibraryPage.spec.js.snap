// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellLibraryPage /> renders this UI as expected 1`] = `
<section
  className="SpellLibraryPage"
>
  <div
    className="SpellLibraryPage__container"
  >
    <h2>
      View All Spells
    </h2>
    <div
      className="SpellLibraryOptions__container"
    >
      <label
        htmlFor="class-select"
      >
        Select a class
      </label>
      <select
        id="class-select"
        name="classes"
      >
        <option
          value=""
        >
          Class Name
        </option>
        <option
          value="bard"
        >
          Bard
        </option>
        <option
          value="cleric"
        >
          Cleric
        </option>
        <option
          value="druid"
        >
          Druid
        </option>
        <option
          value="paladin"
        >
          Paladin
        </option>
        <option
          value="sorcerer"
        >
          Sorcerer
        </option>
        <option
          value="wizard"
        >
          Wizard
        </option>
        <option
          value="warlock"
        >
          Warlock
        </option>
      </select>
      <label
        htmlFor="spell-filter"
      >
        Filter
      </label>
      <select
        id="spell-filter"
        name="filters"
      >
        <option
          value=""
        >
          By
        </option>
        <option
          value="level"
        >
          Level
        </option>
        <option
          value="range"
        >
          Range
        </option>
        <option
          value="TESTVLAUES"
        >
          TEST VALUES
        </option>
        <option
          value="TESTVLAUES2"
        >
          TEST VALUES 2
        </option>
      </select>
    </div>
    <div
      className="SpellLibraryPage__Spells__container"
    />
  </div>
</section>
`;
